{
    "CVE_data_meta": {
        "ASSIGNER": "security-advisories@github.com",
        "ID": "CVE-2021-32688",
        "STATE": "PUBLIC",
        "TITLE": "Application specific tokens can change their own scope"
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "product": {
                        "product_data": [
                            {
                                "product_name": "security-advisories",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_value": "< 19.0.13"
                                        },
                                        {
                                            "version_value": ">= 20.0.0, < 20.0.11"
                                        },
                                        {
                                            "version_value": ">= 21.0.0, < 21.0.3"
                                        }
                                    ]
                                }
                            }
                        ]
                    },
                    "vendor_name": "nextcloud"
                }
            ]
        }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "Nextcloud Server is a Nextcloud package that handles data storage. Nextcloud Server supports application specific tokens for authentication purposes. These tokens are supposed to be granted to a specific applications (e.g. DAV sync clients), and can also be configured by the user to not have any filesystem access. Due to a lacking permission check, the tokens were able to change their own permissions in versions prior to 19.0.13, 20.0.11, and 21.0.3. Thus fileystem limited tokens were able to grant themselves access to the filesystem. The issue is patched in versions 19.0.13, 20.0.11, and 21.0.3. There are no known workarounds aside from upgrading."
            }
        ]
    },
    "impact": {
        "cvss": {
            "attackComplexity": "LOW",
            "attackVector": "NETWORK",
            "availabilityImpact": "HIGH",
            "baseScore": 8.8,
            "baseSeverity": "HIGH",
            "confidentialityImpact": "HIGH",
            "integrityImpact": "HIGH",
            "privilegesRequired": "LOW",
            "scope": "UNCHANGED",
            "userInteraction": "NONE",
            "vectorString": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H",
            "version": "3.1"
        }
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "CWE-285: Improper Authorization"
                    }
                ]
            }
        ]
    },
    "references": {
        "reference_data": [
            {
                "name": "https://github.com/nextcloud/security-advisories/security/advisories/GHSA-48m7-7r2r-838r",
                "refsource": "CONFIRM",
                "url": "https://github.com/nextcloud/security-advisories/security/advisories/GHSA-48m7-7r2r-838r"
            },
            {
                "name": "https://github.com/nextcloud/server/pull/27000",
                "refsource": "MISC",
                "url": "https://github.com/nextcloud/server/pull/27000"
            },
            {
                "name": "https://hackerone.com/reports/1193321",
                "refsource": "MISC",
                "url": "https://hackerone.com/reports/1193321"
            },
            {
                "refsource": "FEDORA",
                "name": "FEDORA-2021-9b421b78af",
                "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/J63NBVPR2AQCAWRNDOZSGRY5II4WS2CZ/"
            },
            {
                "refsource": "FEDORA",
                "name": "FEDORA-2021-6f327296fe",
                "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BVZS26RDME2DYTKET5AECRIZDFUGR2AZ/"
            },
            {
                "refsource": "GENTOO",
                "name": "GLSA-202208-17",
                "url": "https://security.gentoo.org/glsa/202208-17"
            }
        ]
    },
    "source": {
        "advisory": "GHSA-48m7-7r2r-838r",
        "discovery": "UNKNOWN"
    }
}